frontend/src/modules/positions/components/NewPosition.jsx

```diff
@@ -1,42 +1,164 @@
-    state = newPositionFields.reduce((acc, cur) => ({ ...acc, [cur.value]: "" }), {})
+    componentDidMount() {
+        this.props.fetchInstructors()
+    }
+    state = { ...initialState }
     handleChange = prop => event => this.setState({ [prop]: event.target.value })
-    handleSubmit = () => {
+
+    handleSubmit = (e) => {
+        e.preventDefault()
         this.props.createNewPosition(this.state)
     }
+    getInvalid = () =>
+        newPositionFields.reduce((acc, cur) => {
+            if (acc) {
+                return acc
+            }
+            const errorMessage = cur.validate.reduce(
+                (running, validator) => running || validator(this.state[cur.value], this.props),
+                false
+            )
+            return errorMessage ? `${cur.label}: ${errorMessage}` : acc
+        }, false)
+    setInstructor = name => this.setState({ instructor: name })
+    
     render() {
+        const invalid = this.getInvalid()
         return (
             <Grid>
                 <Col xs={8} xsOffset={2}>
                     <h2>New Position</h2>
-                    {newPositionFields.map(({ label, value }) => (
-                        <div key={value}>
-                            <label>{label}</label>
-                            <input value={this.state[value]} onChange={this.handleChange(value)} />
-                        </div>
-                    ))}
-                    <Button bsStyle="primary" onClick={this.handleSubmit}>
-                        Save
-                    </Button>
+                    <form onSubmit={this.handleSubmit}>
+                        {newPositionFields.map(({ value, label, required }) => {
+                            switch (value) {
+                                case 'instructor':
+                                    return <div>
+                                            <InstructorInput 
+                                                curValue={this.state[value]} 
+                                                required={required}
+                                                key={label} 
+                                                label={label} 
+                                                value={value} 
+                                                onChange={this.handleChange(value)}
+                                                />
+                                            { this.state.instructor !== '' &&
+                                            <InstructorList 
+                                                curValue={this.state[value]} 
+                                                setInstructor={this.setInstructor} 
+                                                key='instructorList' 
+                                                instructors={this.props.instructors} /> }
+                                        </div>
+
+                                case 'qualifications':              
```

Comment:
is this needed ?

Reply:
In switch statements, when a `case` is matched, it runs what's in the `case`, and all following statements as well. So in this case, it will run the `return <TextboxInput>` block for both qualifications and duties.

More info / live example of an empty case clause here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
Ah, I see, thanks for clarifying.